DuckDuckGo/AppLifecycle/AppStates/Background.swift

```diff
@@ -18,15 +18,67 @@
 //
 
 import UIKit
+import Core
 
 struct Background: AppState {
 
+    let timestamp = Date()
+
     init(application: UIApplication) {
 
     }
 
 }
 
 struct DoubleBackground: AppState {
-    
+
+    private let dateFormatter: ISO8601DateFormatter = {
+        let formatter = ISO8601DateFormatter()
+        formatter.formatOptions = [.withInternetDateTime]
+        return formatter
+    }()
+
+    let firstTimeBackgroundTimestamp: Date
+    var counter: Int
+
+    init(firstTimeBackgroundTimestamp: Date, counter: Int) {
+        let secondBackgroundTimestamp = Date()
+        self.firstTimeBackgroundTimestamp = secondBackgroundTimestamp
```

Comment:
These names are difficult to reason about, maybe:

`let firstTimeBackgroundTimestamp: Date` -> `let currentBackgroundEventStartTimestamp: Date`

`init(prevoiusDidEnterBackgroundTimestamp: Date, counter: Int)`